src/modules/ekf2/EKF/gps_yaw_fusion.cpp

```diff
@@ -41,99 +41,37 @@
  */
 
 #include "ekf.h"
+#include "python/ekf_derivation/generated/compute_gnss_yaw_innon_innov_var_and_h.h"
 
 #include <mathlib/mathlib.h>
 #include <cstdlib>
 
 void Ekf::fuseGpsYaw(const gpsSample& gps_sample)
 {
-	// assign intermediate state variables
-	const float q0 = _state.quat_nominal(0);
-	const float q1 = _state.quat_nominal(1);
-	const float q2 = _state.quat_nominal(2);
-	const float q3 = _state.quat_nominal(3);
-
 	// calculate the observed yaw angle of antenna array, converting a from body to antenna yaw measurement
 	const float measured_hdg = wrap_pi(gps_sample.yaw + _gps_yaw_offset);
 
-	// define the predicted antenna array vector and rotate into earth frame
-	const Vector3f ant_vec_bf = {cosf(_gps_yaw_offset), sinf(_gps_yaw_offset), 0.0f};
-	const Vector3f ant_vec_ef = _R_to_earth * ant_vec_bf;
-
-	// calculate predicted antenna yaw angle
-	const float predicted_hdg = atan2f(ant_vec_ef(1), ant_vec_ef(0));
-
-	// wrap the innovation to the interval between +-pi
-	const float heading_innov = wrap_pi(predicted_hdg - measured_hdg);
-
 	// using magnetic heading process noise
 	// TODO extend interface to use yaw uncertainty provided by GPS if available
 	const float R_YAW = sq(fmaxf(_params.gps_heading_noise, 1.0e-2f));
 
+	float heading_innov;
+	float heading_innov_var;
+	Vector24f H;
+
+	sym::ComputeGnssYawInnonInnovVarAndH(getStateAtFusionHorizonAsVector(), P, _gps_yaw_offset, measured_hdg, R_YAW, FLT_EPSILON, &heading_innov, &heading_innov_var, &H);
```

Comment:
Do you know why we don't already use the reported heading accuracy? 

https://github.com/PX4/PX4-GPSDrivers/blob/b49a6c257371abae6e26e0a3e0fc04b963b2f13d/src/ubx.cpp#L2032

Reply:
Probably because that information wasn't available when the fusion was implemented the first time. But I can add it now.

Reply:
done in [d91b24f](https://github.com/PX4/PX4-Autopilot/pull/20635/commits/d91b24f8933723862a662ec2b93cf5dd56469555)